### PR TITLE
Add support for installing 0.4.x version of Solidity

### PR DESCRIPTION
**Pull request by ChrisChinchilla**

### CODE REVIEW

**Comment by ChrisChinchilla** [comment]
So we can trigger tests (which passed), I created a new PR here that combines https://github.com/ethereum/homebrew-ethereum/pull/219 and https://github.com/ethereum/homebrew-ethereum/pull/212

**Comment by ChrisChinchilla** [comment]
Merged everything together to one tidy(ish) PR @chriseth and @axic

File: solidity@4.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class SolidityAT4 < Formula
+  desc "The Solidity Contract-Oriented Programming Language"
+  homepage "http://solidity.readthedocs.org"
+  url "https://github.com/ethereum/solidity/releases/download/v0.4.25/solidity_0.4.25.tar.gz"
+  version "0.4.25"
+  sha256 "8172c126973eae1d9d6c7b5071910a9c7475ac64df3945f755cee66104add72d"
+
+  depends_on "cmake" => :build
+  depends_on "boost@1.60" => "c++11"
```
Comment (axic) [inline solidity@4.rb]: Why is this needing a specific version while `0.5.x` doesn't?

Comment (ChrisChinchilla) [reply solidity@4.rb]: This gets a little beyond my C++ knowledge, but iirc, something changed with the boost versions used recently, and if you build this version of Solidity with the latest version of boost, you get errors. When I researched those errors, it seemed that the change happened in recent boost versions, this is the closest Homebrew version of boost available that works with compiling this version of Solidity. You can try it if you like and see what I mean, but this fixed those build errors.

**Comment by ChrisChinchilla** [comment]
@axic Answered, and I'll update docs and issues accordingly when this is merged.

File: solidity@4.rb (hunk 1 above)
Comment (axic) [reply solidity@4.rb]: @ekpyron @chriseth do you happen to know the reason? If we release 0.4.26 for fixing the Firefox inconvenience, we may as well fix this one.

Comment (chriseth) [reply solidity@4.rb]: I don't know - 1.60 sounds a little old, too...

Comment (axic) [reply solidity@4.rb]: Well the Solidity cmake does check for the minimum supported version so this should be safe.

Comment (axic) [reply solidity@4.rb]: > 1.60 sounds a little old, too...

Keep in mind this is compiling 0.4.x which assumed 1.54 as a minimum.

**Comment by axic** [comment]
I'm merging this to have some progress. Created an issue to track: https://github.com/ethereum/solidity/issues/6349

**Comment by zulhfreelancer** [comment]
Thank you @ChrisChinchilla for this PR!

I wrote a guide here to share my steps how the downgrade process from 0.5.x to 0.4.x looks like.

[https://link.medium.com/TLnfCI48fV](https://link.medium.com/TLnfCI48fV)

File: solidity@4.rb (hunk 1 above)
Comment (ChrisChinchilla) [reply solidity@4.rb]: @chriseth and @axic These are sadly the version available - https://formulae.brew.sh/formula/boost

I couldn't get a build to work with the 1.69 version, but I am not sure why, I can share logs if need be. Or we know of a specific boost version that is newer, and does work, i can make a PR to their formula?

**Pull request merged**